hotel/tests/test_hotel.py

```diff
@@ -0,0 +1,264 @@
+from openerp.tests.common import TransactionCase
+from openerp.tools import mute_logger
+from openerp.tools import float_round
+from openerp.exceptions import ValidationError
+from openerp.exceptions import except_orm
+
+
+class TestHotel(TransactionCase):
+    """Tests hotel management addon"""
+
+    def setUp(self):
+        super(TestHotel, self).setUp()
+        self.folioObj = self.env['hotel.folio']
+        self.saleOrderObj = self.env['sale.order']
+        self.saleOrderLineObj = self.env['sale.order.line']
+        self.invoiceObj = self.env['account.invoice']
+        self.voucherObj = self.env['account.voucher']
+        self.roomlineObj = self.env['hotel.folio.room']
+        self.servicelineObj = self.env['hotel.folio.service']
+        self.ModelDataObj = self.env['ir.model.data']
+        self.partnerObj = self.env['res.partner']
+
+        # ----------------------------------------------------------------------
+        # Getting data for two users
+        # ----------------------------------------------------------------------
+        self.admin = self.partnerObj.browse(3)
+        self.testUser = self.partnerObj.browse(1)
+
+        # ----------------------------------------------------------------------
+        # Getting data for two rooms
+        # ----------------------------------------------------------------------
+        self.room1_id = self.ModelDataObj.xmlid_to_res_id('hotel.hotel_room_5')
+        self.room1 = self.env['hotel.room'].browse(self.room1_id)
+        self.room2_id = self.ModelDataObj.xmlid_to_res_id('hotel.hotel_room_2')
+        self.room2 = self.env['hotel.room'].browse(self.room2_id)
+
+        # ----------------------------------------------------------------------
+        # Getting servie data
+        # ----------------------------------------------------------------------
+        self.service1_id = self.ModelDataObj.xmlid_to_res_id(
+            'hotel.hotel_service_6'
+        )
+        self.service1 = self.env['hotel.services'].browse(self.service1_id)
+
+    @mute_logger('openerp.addons.base.ir.ir_model', 'openerp.models')
+    def test_folio_create(self):
+        '''Creates a new folio'''
+        # ----------------------------------------------------------------------
+        # Creating a new folio
+        # ----------------------------------------------------------------------
+        folio0 = self.folioObj.create({
+            'partner_id': self.admin.id,
+            'partner_invoice_id': self.admin.id,
+            'partner_shipping_id': self.admin.id,
+            'checkin_date': '2015-07-20 11:04:38',
```

Comment:
Should you be able to create a folio in the past?
If not then I would rather set the checking date to "now + 1 day" in this test.